File: src/scope-type.ts
export const SCOPE_TYPE = {
  Block: "block",
  Catch: "catch",
  Class: "class",
  ClassFieldInitializer: "class-field-initializer",
  ClassStaticBlock: "class-static-block",
  For: "for",
  Function: "function",
  FunctionExpressionName: "function-expression-name",
  Global: "global",
  Module: "module",
  Switch: "switch",
  With: "with",
} as const;
export type ScopeType = (typeof SCOPE_TYPE)[keyof typeof SCOPE_TYPE];

File: crates/unsnarl-ir/src/scope_type.rs
//! Scope categorization. Ports `ts/src/analyzer/scope-type.ts`.
//!
//! Lives in `unsnarl-ir` rather than `unsnarl-analyzer` because the IR
//! contract types reference it and `unsnarl-ir` is below
//! `unsnarl-analyzer` in the dependency graph.

use serde::Serialize;

#[derive(Serialize)]
#[serde(rename_all = "kebab-case")]
pub enum ScopeType {
    Block,
    Catch,
    Class,
    ClassFieldInitializer,
    ClassStaticBlock,
    For,
    Function,
    FunctionExpressionName,
    Global,
    Module,
    Switch,
    With,
}
